ipatests/test_webui/test_cert.py

```diff
@@ -23,10 +23,93 @@
 
 from ipatests.test_webui.ui_driver import UI_driver
 from ipatests.test_webui.ui_driver import screenshot
+from datetime import date, timedelta
 import pytest
 
 ENTITY = 'cert'
 
+CERT_CSR = (""""-----BEGIN NEW CERTIFICATE REQUEST-----
+MIICfDCCAWQCAQAwNzEWMBQGA1UEChMNVEVTVFJFTE0uVEVTVDEdMBsGA1UEAxMU
+bWFzdGVyLnRlc3RyZWxtLnRlc3QwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEK
+AoIBAQC1tciFuG4ri2XUfY/QmyiA9wNECUWe4BvKe6+HgGj70nG1V5IRLFLUDKxk
+txxIw4W75oQyS4M26cHj8PhyIlnFyhahjWccd66Mij9EWjYLprvsZUcV8t6hfQN7
+a4CZxQ1fe2yEmw4tfT5wKFr1trmdTITaUzcs4f0fVzbTLA5Ird84UJEDFLWJIcUu
+T08I0QPoc2Xd3Zou/olSyF8kBem4LheYaAYQGO8k568oVDOoAB8CNIrANe1GwS7U
+dAjCdOKyJba/tYaC8wuXZAtFFrOwUKq7oTWasdC7jctOwSXz8dbKp3DkqrHRHOOV
+ZMD/h6MBdeYRC7qNucBi7kV/vgetAgMBAAGgADANBgkqhkiG9w0BAQsFAAOCAQEA
+LUgNGI2VGZZPqW0ChDnrG/LxrLxpQF3HkuXblfPlMX9/zTKG4DwSYDN/hoKtuCti
+q9qauq3BRoPkN0gcZGplq8inDP1p9UQOBPF1107ZODOt3W1vjBZ6hziZVbo6w5B3
+kJeLinP7TGZ6SjnEamEJT+WX0LkfxGyficNDs5lb0jaRmY/2aqvP3Tmx924JAzfo
+6x673lBidFz1CxaRAxgV7O50jWt4BdvaPVtcG6H1l3GQrzQUD+Zc70QhTI/3eGJg
+rzgBjBS9cdroNHIJt7Z1CWeFNWzgiH367962uxTdcoVy9Tkx5+9On8muHcMYUl1d
+kkE7vpltl6NGh0qqfrm2VA==
+-----END NEW CERTIFICATE REQUEST-----""")
+
+ERR_SPACE = "invalid '{}': Leading and trailing spaces are not allowed"
+ERR_MUST_INTEGER = "invalid '{}': must be an integer"
+LEAST_SERIAL = "invalid '{}': must be at least 0"
+INV_DATE = ("invalid '{}': does not match any of accepted formats: "
+            "%Y%m%d%H%M%SZ, %Y-%m-%dT%H:%M:%SZ, %Y-%m-%dT%H:%MZ, "
+            "%Y-%m-%dZ, %Y-%m-%d %H:%M:%SZ, %Y-%m-%d %H:%MZ")
+
+
+def search_pkey(self, pkey):
+    search_field_s = '.search-filter input[name=filter]'
+    self.fill_text(search_field_s, pkey)
+    self.action_button_click('find', parent=None)
+    self.wait_for_request(n=2)
+
+
+def add_cert(self, principal, csr):
+    self.facet_button_click('request_cert')
+    self.fill_textbox('principal', principal)
+    self.check_option('add', 'checked')
+    self.fill_textarea('csr', csr)
+    self.dialog_button_click('issue')
+    self.navigate_to_entity(ENTITY)
+    rows = self.get_rows()
+    return rows[-1]
+
+
+def revoke_cert(self, record, reason):
+    self.navigate_to_entity(ENTITY)
+    self.navigate_to_row_record(record)
+    self.action_list_action('revoke_cert', False)
+    self.select('select[name=revocation_reason]', reason)
+    self.dialog_button_click('ok')
+
+
+def check_option_negative(self, date, option):
+    self.navigate_to_entity(ENTITY)
+    self.select('select[name=search_option]', option)
+    search_pkey(self, date)
+    self.assert_last_error_dialog(INV_DATE.format(option))
+    self.close_all_dialogs()
+
+
+def check_space_error(self, string, option):
+    self.navigate_to_entity(ENTITY)
+    self.select('select[name=search_option]', option)
+    search_pkey(self, string)
+    self.assert_last_error_dialog(ERR_SPACE.format(option))
+    self.close_all_dialogs()
+
+
+def check_integer(self, string, option):
```

Comment:
I would recommend to add a docstring here as "check_integer" is not much descriptive from first look.